FinancialTracker/ViewControllers/BalanceViewController.swift

```diff
@@ -56,28 +59,39 @@ class BalanceViewController: UIViewController {
     
     @IBAction func nextButtonTapped(_ sender: Any) {
         guard let balance = balanceTextField.text?.trimmingCharacters(in: .whitespacesAndNewlines), !balance.isEmpty else {
-            self.present(UIAlertController.create(title: "Missing Balance", message: "Please fill in your starting balance"), animated: true)
+            present(UIAlertController.create(title: "Missing Balance", message: "Please fill in your starting balance"), animated: true)
             return
         }
         
         guard let balanceNumber = Double(balance) else {
-            self.present(UIAlertController.create(title: "Invalid Format", message: "Please fill in a number"), animated: true)
+            present(UIAlertController.create(title: "Invalid Format", message: "Please fill in a number"), animated: true)
             return
         }
         
         let selectedCurrency = currencies[currencyPicker.selectedRow(inComponent: 0)]
         
-        authManager?.addBalanceToCurrentUser(balanceNumber, currency: selectedCurrency) { firebaseError, _ in
-            switch firebaseError {
-            case .access(let error):
-                guard let error = error else { return }
-                self.present(UIAlertController.create(title: "Access Error", message: error), animated: true)
-            case .auth, .database, .unknown, .signOut, .nonExistingUser:
-                // swiftlint:disable:next force_unwrapping
-                assertionFailure("This error should not appear: \(firebaseError!.localizedDescription)")
-                // swiftlint:disable:next unneeded_break_in_switch
-                break
-            case .none:
+        authManager?.addBalanceToCurrentUser(balanceNumber, currency: selectedCurrency) { authError, _ in
+            if let authError = authError {
+                switch authError {
+                case .database(let error):
+                    if let databaseError = error {
+                        switch databaseError {
+                        case .database(let error):
```

Comment:
Here you can optimise all these lines of code with an extension for the `UIAlertController` which takes say an `AuthError` in it's `init` and gives you back and `alert` based on that.